json-path/src/test/java/io/zeebe/msgpack/mapping/MappingExtractParameterizedTest.java

```diff
@@ -287,6 +288,38 @@
             createMapping("$['foo']['bar']", "$.result"),
             // expected result
             "{'result':1}"
+          },
+          {
+            // source
+            "{'key':'val'}",
+            // mapping
+            createMapping("$.notAKey", "$.key"),
+            // expected result
+            "{'key':null}"
+          },
+          {
+            // source
+            "{'key':'val'}",
+            // mapping
+            createMapping("$.notAKey", "$.arr", Type.COLLECT),
+            // expected result
+            "{'arr':[null]}"
+          },
+          {
+            // source
+            "{'key1':'val1', 'key2': 'val2'}",
+            // mapping
+            createMapping("$.*", "$.newKey", Type.PUT),
```

Comment:
What happens on collect?

Reply:
I think it would be nice if all the results were collected in an array in this case, but as this is not straightforward to add, I will make it collect only the first result for now and add a test. Let's see if this is actually a use case.